src/passes/GlobalTypeOptimization.cpp

```diff
@@ -177,8 +177,8 @@ struct GlobalTypeOptimization : public Pass {
         continue;
       }
       auto& fields = type.getStruct().fields;
-      auto& subSuper = subSupers[type];
-      auto& sub = subs[type];
+      auto& dataFromSubsAndSupers = dataFromSubsAndSupersMap[type];
+      auto& dataFromSupers = dataFromSupersMap[type];
```

Comment:
We probably don't need the "dataFrom" prefix if you want to tighten this up. No strong feeling either way, though.